tpm2/tpm2_test.go

```diff
@@ -1363,3 +1363,47 @@ func TestCreatePrimaryRawTemplate(t *testing.T) {
 		t.Errorf("got key exponent %v, want %v", pubRSA.E, defaultKeyParams.RSAParameters.Exponent)
 	}
 }
+
+func TestMatchesTemplate(t *testing.T) {
+	makePublic := func() Public {
+		return Public{
+			Type:       AlgRSA,
+			NameAlg:    AlgSHA256,
+			Attributes: FlagSignerDefault,
+			RSAParameters: &RSAParams{
+				Sign: &SigScheme{
+					Alg:  AlgRSASSA,
+					Hash: AlgSHA256,
+				},
+				KeyBits:  2048,
+				Exponent: 0,
```

Comment:
Nit: the `Exponent` field will never contain zero in a `Public` that was wire decoded - it will always be rewritten to `defaultRSAExponent` and have the `encodeDefaultExponentAsZero` bit set. Maybe we should test using the `defaultRSAExponent` value?

Reply:
or test both?